src/Api/include/Api/Event.h

```diff
@@ -0,0 +1,209 @@
+  ApiEncodedString(const char* name) { EncodeString(name, this); }
+  void set_encoded_name_1(uint64_t value) { encoded_name_1 = value; }
+  void set_encoded_name_2(uint64_t value) { encoded_name_2 = value; }
+  void set_encoded_name_3(uint64_t value) { encoded_name_3 = value; }
+  void set_encoded_name_4(uint64_t value) { encoded_name_4 = value; }
+  void set_encoded_name_5(uint64_t value) { encoded_name_5 = value; }
+  void set_encoded_name_6(uint64_t value) { encoded_name_6 = value; }
+  void set_encoded_name_7(uint64_t value) { encoded_name_7 = value; }
+  void set_encoded_name_8(uint64_t value) { encoded_name_8 = value; }
+  void add_encoded_name_additional(uint64_t value) { encoded_name_additional.push_back(value); }
+  uint64_t encoded_name_1{};
+  uint64_t encoded_name_2{};
+  uint64_t encoded_name_3{};
+  uint64_t encoded_name_4{};
+  uint64_t encoded_name_5{};
+  uint64_t encoded_name_6{};
+  uint64_t encoded_name_7{};
+  uint64_t encoded_name_8{};
+  std::vector<uint64_t> encoded_name_additional{};
+};
+
+struct ApiScopeStart {
+  ApiScopeStart(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name,
+                orbit_api_color color_rgba = kOrbitColorAuto, uint64_t group_id = 0,
+                uint64_t address_in_function = 0)
+      : meta_data(pid, tid, timestamp_ns),
+        group_id(group_id),
+        address_in_function(address_in_function),
+        encoded_name(name),
+        color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t group_id{};
+  uint64_t address_in_function{};
+
+  ApiEncodedString encoded_name;
+
+  uint32_t color_rgba{};
+};
+
+struct ApiScopeStop {
+  ApiScopeStop(int32_t pid, int32_t tid, uint64_t timestamp_ns)
+      : meta_data(pid, tid, timestamp_ns) {}
+
+  ApiEventMetaData meta_data;
+};
+
+struct ApiScopeStartAsync {
+  ApiScopeStartAsync(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name, uint64_t id,
+                     orbit_api_color color_rgba = kOrbitColorAuto, uint64_t address_in_function = 0)
+      : meta_data(pid, tid, timestamp_ns),
+        id(id),
+        address_in_function(address_in_function),
+        encoded_name(name),
+        color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t id{};
+  uint64_t address_in_function{};
+
+  ApiEncodedString encoded_name;
+
+  uint32_t color_rgba{};
+};
+
+struct ApiScopeStopAsync {
+  ApiScopeStopAsync(int32_t pid, int32_t tid, uint64_t timestamp_ns, uint64_t id)
+      : meta_data(pid, tid, timestamp_ns), id(id) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t id{};
+};
+
+struct ApiStringEvent {
+  ApiStringEvent(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name, uint64_t id,
+                 orbit_api_color color_rgba = kOrbitColorAuto)
+      : meta_data(pid, tid, timestamp_ns), id(id), encoded_name(name), color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t id{};
+
+  ApiEncodedString encoded_name;
+  uint32_t color_rgba{};
```

Comment:
Nit. meta_data and encoded_name are one after the other in other structs, maybe do the same here. Also, mega nit, but why the empty lines between the fields? I think it might be easier to read with all the fields tightly packed.

Reply:
I don't get the benefit of adding the color to another struct. The reason for the compositon of structs for meta_data and strings is to have less code duplication, and less lines.

I removed the empty lines.

Reply:
Oh sorry, that wasn't clear, I was only commenting on the order of fields in the struct. Nit picking as mentioned :-)

  ApiEventMetaData meta_data;
  uint64_t id{};
  ApiEncodedString encoded_name;
  uint32_t color_rgba{};

vs 

ApiEventMetaData meta_data;
ApiEncodedString encoded_name;
uint64_t id{};
uint32_t color_rgba{};

Reply:
Done.